src/main/common/time.c

```diff
@@ -98,6 +98,17 @@ static void rtcTimeToDateTime(dateTime_t *dt, rtcTime_t t)
     dt->millis = t % MILLIS_PER_SECOND;
 }
 
+static bool rtcIsDateTimeValid(dateTime_t *dateTime)
+{
+    return (dateTime->year >= UNIX_REFERENCE_YEAR) &&
+           (dateTime->month >= 1 && dateTime->month <= 12) &&
+           (dateTime->day >= 1 && dateTime->day <= 31) &&
+           (dateTime->hours <= 23) &&
+           (dateTime->minutes <= 59) &&
+           (dateTime->seconds <= 59) &&
+           (dateTime->millis <= 999);
+}
+
 static void dateTimeFormat(char *buf, dateTime_t *dateTime, int16_t offset)
```

Comment:
I'd make this return `bool`, as well as the other formatting functions. This way a caller can find out whether the formatting took place.